allensdk/internal/brain_observatory/time_sync.py

```diff
@@ -30,25 +36,85 @@
 
 # MPE is changing keys. This isn't versioned in the file.
 VERSION_2_KEYS = {
-    "photodiode": "photodiode",
+    "photodiode": "stim_photodiode",
     "2p": "2p_vsync",
     "stimulus": "stim_vsync",
     "eye_camera": "eye_tracking",
     "behavior_camera": "behavior_monitoring",
     "acquiring": "2p_acquiring",
     "lick_sensor": "lick_sensor"
     }
+"""
+
+POSSIBLE_KEY_PAIRS = {
+        "photodiode": ("stim_photodiode", "photodiode"),
+        "eye_camera": ("cam2_exposure", "eye_tracking"),
+        "behavior_camera": ("cam1_exposure", "behavior_monitoring"),
+        "lick_sensor": ("lick_1", "lick_sensor")
+        }
+
+
+def validate_keys(key_set: Set, dictionary_with_keys: Dict):
+    """
+    Validates that all values in the dictionary_with_keys are present in
+    the list.
+
+    Args:
+        key_set: Set of key values
+        dictionary_with_keys: Dictionary where values should be contained in
+                              key_list
+
+    Returns:
+        non_matches: returns a list of values not found in the key_set but
+                    present in dictionary
+    """
+    return_keys = []
+    for key, value in dictionary_with_keys.items():
+        if key not in key_set:
+            return_keys.append(key)
+    return return_keys
 
 
 def get_keys(sync_dset):
     """Get the correct lookup for line labels.
 
     This method is fragile, but not all old data contains the full list
     of keys.
+
+    Update (4/24/2020):
+    This function was changed as previous test cases were failing
+    and the error was being handled. When this error handling was removed
+    in the monitor delay refactor (#1438) the tests broke. This method now
+    searches for the possible keys in the set and if they exist adds them to
+    the return dictionary. If keys don't exist it presumes old file and defaults
+    to legacy keys, logging this as a warning. This fixes the bamboo test issues
     """
-    if "cam2_exposure" in sync_dset.line_labels:
-        return VERSION_1_KEYS
-    return VERSION_2_KEYS
+    key_dict = {
+            "photodiode": None,
+            "2p": "2p_vsync",
+            "stimulus": "stim_vsync",
+            "eye_camera": None,
+            "behavior_camera": None,
+            "acquiring": "2p_acquiring",
+            "lick_sensor": None
+            }
+    for key, value in POSSIBLE_KEY_PAIRS.items():
+        if value[0] in sync_dset.line_labels:
+            key_dict[key] = value[0]
+        elif value[1] in sync_dset.line_labels:
+            key_dict[key] = value[1]
+        else:
+            logging.warning(f"No key found in sync dataset line labels for key:"
```

Comment:
you don't need the `f` if there's nothing there to format.

Reply:
my bad that's from splitting the f string, fixed

Reply:
@djkapner should resolve this conversation